### ISSUE TITLE
Node Segmentation fault on Windows Server

### ISSUE THREAD

**Issue by jakubgs** [status-im/nim-beacon-chain#2666]
**Describe the bug**

I'm seeing semi-random Segmentation faults on the Windows node I configured as part of https://github.com/status-im/infra-nimbus/issues/59:
```
nimbus@windows-01 MINGW64 ~/beacon-node-prater-stable                                                                                                                                                                                                                                                 
$ ./bin/nimbus_beacon_node.exe --non-interactive --network=prater --data-dir=C:/Users/nimbus/beacon-node-prater-stable/data/shared_prater_0 --web3-url=wss://goerli.infura.io/ws/v3/6224f3c792cc443fafb64e70a98f871e --nat=extip:35.232.80.244 --log-level=DEBUG --log-file=C:/Users/nimbus/beacon-node
-prater-stable/log/nimbus.log --tcp-port=9000 --udp-port=9000 --max-peers=320 --slashing-db-kind=v2 --insecure-netkey-password=true --subscribe-all-subnets=false --doppelganger-detection=true --rpc --rpc-address=0.0.0.0 --rpc-port=9900 --metrics --metrics-address=0.0.0.0 --metrics-port=9200    
{"lvl":"WRN","ts":"2021-06-22 09:25:05.474+00:00","msg":"The --log-file option is not active in the current build","tid":7680,"file":"nimbus_binary_common.nim:70"}                                                                                                                                   
{"lvl":"INF","ts":"2021-06-22 09:25:05.474+00:00","msg":"Launching beacon node","topics":"beacnde","tid":7680,"file":"nimbus_beacon_node.nim:1672","version":"v1.4.0-44f652-stateofus","bls_backend":"BLST","cmdParams":["--non-interactive","--network=prater","--data-dir=C:/Users/nimbus/beacon-node
-prater-stable/data/shared_prater_0","--web3-url=wss://goerli.infura.io/ws/v3/6224f3c792cc443fafb64e70a98f871e","--nat=extip:35.232.80.244","--log-level=DEBUG","--log-file=C:/Users/nimbus/beacon-node-prater-stable/log/nimbus.log","--tcp-port=9000","--udp-port=9000","--max-peers=320","--slashing
-db-kind=v2","--insecure-netkey-password=true","--subscribe-all-subnets=false","--doppelganger-detection=true","--rpc","--rpc-address=0.0.0.0","--rpc-port=9900","--metrics","--metrics-address=0.0.0.0","--metrics-port=9200"],"config":{"logLevel":"DEBUG","logFile":"C:/Users/nimbus/beacon-node-pra
ter-stable/log/nimbus.log","eth2Network":"prater","dataDir":"C:/Users/nimbus/beacon-node-prater-stable/data/shared_prater_0","validatorsDirFlag":null,"secretsDirFlag":null,"walletsDirFlag":null,"web3Urls":["wss://goerli.infura.io/ws/v3/6224f3c792cc443fafb64e70a98f871e"],"web3Mode":0,"nonInterac
tive":true,"netKeyFile":"random","netKeyInsecurePassword":true,"agentString":"nimbus","subscribeAllSubnets":false,"slashingDbKind":1,"cmd":0,"bootstrapNodes":[],"bootstrapNodesFile":"","listenAddress":"0.0.0.0","tcpPort":9000,"udpPort":9000,"maxPeers":320,"nat":{"hasExtIp":true,"extIp":"35.232.
80.244"},"enrAutoUpdate":false,"weakSubjectivityCheckpoint":null,"finalizedCheckpointState":null,"finalizedCheckpointBlock":null,"nodeName":"","graffiti":null,"verifyFinalization":false,"stopAtEpoch":0,"metricsEnabled":true,"metricsAddress":"0.0.0.0","metricsPort":9200,"statusBarEnabled":true,"
statusBarContents":"peers: $connected_peers;finalized: $finalized_root:$finalized_epoch;head: $head_root:$head_epoch:$head_epoch_slot;time: $epoch:$epoch_slot ($slot);sync: $sync_status|ETH: $attached_validators_balance","rpcEnabled":true,"rpcPort":9900,"rpcAddress":"0.0.0.0","restEnabled":fals
e,"restPort":5052,"restAddress":"127.0.0.1","inProcessValidators":true,"discv5Enabled":true,"dumpEnabled":false,"directPeers":[],"doppelgangerDetection":true}}                                                                                                                                       
{"lvl":"NOT","ts":"2021-06-22 09:25:05.474+00:00","msg":"Starting metrics HTTP server","topics":"beacnde","tid":7680,"file":"nimbus_beacon_node.nim:1684","url":"http://0.0.0.0:9200/metrics"}                                                                                                        
{"lvl":"NOT","ts":"2021-06-22 09:25:07.185+00:00","msg":"New database from snapshot","topics":"beacnde","tid":7680,"file":"blockchain_dag.nim:1103","blockRoot":"8c0ebce4","stateRoot":"895390e9","fork":{"previous_version":"00001020","current_version":"00001020","epoch":0},"validators":210000}   
Segmentation fault
````
Sometimes it happens right away, sometimes it's after a while, 30 seconds, sometimes more. But I've never seen it run for more than a few minutes. All three branches show the same behavior.

**Additional context**
The host is `windows-01.gc-us-central1-a.nimbus.prater` and can be accessed via SSH like other Nimbus nodes.
The configurations are available under `C:\Users\nimbus`:
```
nimbus@windows-01 MINGW64 ~
$ pwd
/c/Users/nimbus

nimbus@windows-01 MINGW64 ~
$ ls -l | grep beacon-node
drwxr-xr-x 1 nimbus 197121      0 Jun 21 09:39 beacon-node-prater-stable
drwxr-xr-x 1 nimbus 197121      0 Jun 21 14:57 beacon-node-prater-testing
drwxr-xr-x 1 nimbus 197121      0 Jun 21 10:27 beacon-node-prater-unstable
```
You can see the service configuration in the YAML file:
```
nimbus@windows-01 MINGW64 ~
$ cd beacon-node-prater-stable/

nimbus@windows-01 MINGW64 ~/beacon-node-prater-stable
$ ls -l
total 11481
-rw-r--r-- 1 nimbus 197121        0 Jun 21 09:33 beacon-node-prater-stable.err.log
-rwxr-xr-x 1 nimbus 197121   655872 Jun 21 09:21 beacon-node-prater-stable.exe
-rw-r--r-- 1 nimbus 197121      422 Jun 21 09:51 beacon-node-prater-stable.out.log
-rw-r--r-- 1 nimbus 197121     4315 Jun 21 09:51 beacon-node-prater-stable.wrapper.log
-rw-r--r-- 1 nimbus 197121     1207 Jun 21 09:39 beacon-node-prater-stable.yml
drwxr-xr-x 1 nimbus 197121        0 Jun 21 09:21 bin
-rw-r--r-- 1 nimbus 197121 11066394 Jun 22 01:35 build.log
-rwxr-xr-x 1 nimbus 197121     1885 Jun 21 08:47 build.sh
drwxr-xr-x 1 nimbus 197121        0 Jun 22 09:25 data
drwxr-xr-x 1 nimbus 197121        0 Jun 21 09:21 log
drwxr-xr-x 1 nimbus 197121        0 Jun 22 01:07 repo

nimbus@windows-01 MINGW64 ~/beacon-node-prater-stable
$ grep executable beacon-node-prater-stable.yml
executable: 'C:/Users/nimbus/beacon-node-prater-stable/bin/nimbus_beacon_node.exe'
```

You can poke me directly for mode details of the setup as the docs are not read.

**Comment by jakubgs** [comment]
Sometimes it even manages to connect to a few peers before it explodes:
```
{"lvl":"DBG","ts":"2021-06-22 09:48:53.077+00:00","msg":"Requesting blocks from peer","topics":"beacnde","tid":8068,"file":"sync_manager.nim:756","peer":"16U*XjdNBb","slot":283,"slot_count":32,"step":1,"peer_score":300,"peer_speed":163115.0}
{"lvl":"DBG","ts":"2021-06-22 09:48:53.077+00:00","msg":"decodeMsg: decoded message","topics":"libp2p identify","tid":8068,"file":"identify.nim:102","pubkey":"secp25...71])))","addresses":"@[/ip4/0.0.0.0/tcp/9000]","protocols":"@[\"/ipfs/id/1.0.0\", \"/meshsub/1.1.0\", \"/eth2/beacon_chain/req/
status/1/ssz_snappy\", \"/eth2/beacon_chain/req/ping/1/ssz_snappy\", \"/eth2/beacon_chain/req/metadata/1/ssz_snappy\", \"/eth2/beacon_chain/req/beacon_blocks_by_range/1/ssz_snappy\", \"/eth2/beacon_chain/req/beacon_blocks_by_root/1/ssz_snappy\", \"/eth2/beacon_chain/req/goodbye/1/ssz_snappy\"]"
,"observable_address":"Some(/ip4/35.232.80.244/tcp/53034)","proto_version":"Some(\"ipfs/0.1.0\")","agent_version":"Some(\"nimbus\")"}                                                                                                                                                                 
{"lvl":"DBG","ts":"2021-06-22 09:48:53.143+00:00","msg":"Dial successful","topics":"libp2p switch","tid":8068,"file":"switch.nim:156","conn":"16U*zDNnkH:60d1b1fd1324000000002708","peerInfo":{"peerId":"16Uiu2HAmLo3dfHcGGkdkWocnPHvWjjAAFGX4TKRUPgHQLdzDNnkH","addrs":["/ip4/0.0.0.0/tcp/9000"],"prot
ocols":["/ipfs/id/1.0.0","/meshsub/1.1.0","/eth2/beacon_chain/req/status/1/ssz_snappy","/eth2/beacon_chain/req/ping/1/ssz_snappy","/eth2/beacon_chain/req/metadata/1/ssz_snappy","/eth2/beacon_chain/req/beacon_blocks_by_range/1/ssz_snappy","/eth2/beacon_chain/req/beacon_blocks_by_root/1/ssz_snapp
y","/eth2/beacon_chain/req/goodbye/1/ssz_snappy"],"protoVersion":"ipfs/0.1.0","agentVersion":"nimbus"}}                                                                                                                                                                                               
 peers: 8 ❯ finalized: 8c0ebce4:0 ❯ head: fb7ac3a4:1:24 ❯ time: 20435:24 (653944) ❯ sync: PDDDDDDwww:7:1.3258:1.3258:05d17h00m (27)                                                                                                                                                             ETH: 0 
Segmentation fault
```

**Comment by jakubgs** [comment]
It's worth noting that this uses the MinGW installed by [scoop](https://scoop.sh/):
https://github.com/ScoopInstaller/Main/blob/master/bucket/gcc.json

**Comment by jakubgs** [comment]
I've tried building the `stable` branch with `-d:disableMarchNative` added to `NIMFLAGS` but it also explodes.

**Comment by zah** [comment]
Please note that the build here is executed through the following script:

https://github.com/status-im/infra-role-beacon-node-windows/blob/1809e84d48d31cac0b783ed3a2182fdd23e34779/templates/build.sh.j2#L21-L23

This effectively depends on the compilers installed in the environment (gcc/mingw/etc). We can consider using `make dist` instead, but the compete DevOps setup benefits from the assumption that the binaries will be built locally (on Windows). Perhaps we can run the `make dist` machinery with docker on Windows, @stefantalpalaru?

**Comment by stefantalpalaru** [comment]
Isn't it easier to use MSYS2?

**Comment by jakubgs** [comment]
It is available from `scoop`, I can try using it:
```
admin@windows-01 MINGW64 ~                                                                                                                  
$ scoop search msys2                                                                                                                        
'main' bucket:                                                                                                                              
    msys2 (2021-06-04) 
```

**Comment by zah** [comment]
@stefantalpalaru suggested that we can rule out the service wrapper as a potential culprit by either:

1) Try to run the locally built binaries as stand-alone executables in a regular shell.
2) Try to use the official windows 1.4.0 binaries on the same machine (the other suspect still being the local gcc/mingw version)

**Comment by jakubgs** [comment]
>suggested that we can rule out the service wrapper as a potential culprit

If you look at the log in the issue description no service wrapper is being used.

I can try testing the official binaries later today.

**Comment by jakubgs** [comment]
Nope, it explodes:
```
nimbus@windows-01 MINGW64 ~/beacon-node-prater-stable                                                                                                                                  
$ ./nimbus-eth2_Windows_amd64_1.4.0_44f652f7/build/nimbus_beacon_node.exe --non-interactive --network=prater --data-dir=C:/Users/nimbus/beacon-node-prater-stable/data/shared_prater_0 -
-web3-url=wss://goerli.infura.io/ws/v3/6224f3c792cc443fafb64e70a98f871e --nat=extip:35.232.80.244 --log-level=DEBUG --log-file=C:/Users/nimbus/beacon-node-prater-stable/log/nimbus.log 
--tcp-port=9000 --udp-port=9000 --max-peers=320 --slashing-db-kind=v2 --insecure-netkey-password=true --subscribe-all-subnets=false --doppelganger-detection=true --rpc --rpc-address=0.
0.0.0 --rpc-port=9900 --metrics --metrics-address=0.0.0.0 --metrics-port=9200                                                                                                          
WRN 2021-06-22 12:24:39.643+00:00 The --log-file option is not active in the current build tid=4808 file=nimbus_binary_common.nim:70                                                   
INF 2021-06-22 12:24:39.643+00:00 Launching beacon node                      topics="beacnde" tid=4808 file=nimbus_beacon_node.nim:1672 version=v1.4.0-44f652-stateofus bls_backend=BLST
 cmdParams="@[\"--non-interactive\", \"--network=prater\", \"--data-dir=C:/Users/nimbus/beacon-node-prater-stable/data/shared_prater_0\", \"--web3-url=wss://goerli.infura.io/ws/v3/6224
f3c792cc443fafb64e70a98f871e\", \"--nat=extip:35.232.80.244\", \"--log-level=DEBUG\", \"--log-file=C:/Users/nimbus/beacon-node-prater-stable/log/nimbus.log\", \"--tcp-port=9000\", \"--
udp-port=9000\", \"--max-peers=320\", \"--slashing-db-kind=v2\", \"--insecure-netkey-password=true\", \"--subscribe-all-subnets=false\", \"--doppelganger-detection=true\", \"--rpc\", \
"--rpc-address=0.0.0.0\", \"--rpc-port=9900\", \"--metrics\", \"--metrics-address=0.0.0.0\", \"--metrics-port=9200\"]" config="(logLevel: \"DEBUG\", logFile: Some(C:/Users/nimbus/beaco
n-node-prater-stable/log/nimbus.log), eth2Network: Some(\"prater\"), dataDir: C:/Users/nimbus/beacon-node-prater-stable/data/shared_prater_0, validatorsDirFlag: None[InputDir], secrets
DirFlag: None[InputDir], walletsDirFlag: None[InputDir], web3Urls: @[\"wss://goerli.infura.io/ws/v3/6224f3c792cc443fafb64e70a98f871e\"], web3Mode: auto, nonInteractive: true, netKeyFil
e: \"random\", netKeyInsecurePassword: true, agentString: \"nimbus\", subscribeAllSubnets: false, slashingDbKind: v2, cmd: noCommand, bootstrapNodes: @[], bootstrapNodesFile: , listenA
ddress: 0.0.0.0, tcpPort: 9000, udpPort: 9000, maxPeers: 320, nat: (hasExtIp: true, extIp: 35.232.80.244), enrAutoUpdate: false, weakSubjectivityCheckpoint: None[Checkpoint], finalized
CheckpointState: None[InputFile], finalizedCheckpointBlock: None[InputFile], nodeName: \"\", graffiti: None[GraffitiBytes], verifyFinalization: false, stopAtEpoch: 0, metricsEnabled: t
rue, metricsAddress: 0.0.0.0, metricsPort: 9200, statusBarEnabled: true, statusBarContents: \"peers: $connected_peers;finalized: $finalized_root:$finalized_epoch;head: $head_root:$head
_epoch:$head_epoch_slot;time: $epoch:$epoch_slot ($slot);sync: $sync_status|ETH: $attached_validators_balance\", rpcEnabled: true, rpcPort: 9900, rpcAddress: 0.0.0.0, restEnabled: fals
e, restPort: 5052, restAddress: 127.0.0.1, inProcessValidators: true, discv5Enabled: true, dumpEnabled: false, directPeers: @[], doppelgangerDetection: true)"                         
NOT 2021-06-22 12:24:39.723+00:00 Starting metrics HTTP server               topics="beacnde" tid=4808 file=nimbus_beacon_node.nim:1684 url=http://0.0.0.0:9200/metrics                
NOT 2021-06-22 12:24:41.424+00:00 New database from snapshot                 topics="beacnde" tid=4808 file=blockchain_dag.nim:1103 blockRoot=8c0ebce4 stateRoot=895390e9 fork="(previou
s_version: 00001020, current_version: 00001020, epoch: 0)" validators=210000                                                                                                           
Segmentation fault 
```

**Comment by jakubgs** [comment]
Interestingly enough, I ran essentially the same command on a VM running Windows 10, and it seems to work fine:

![image](https://user-images.githubusercontent.com/2212681/122954775-06e6dd80-d380-11eb-80d8-b5412feb41c7.png)

So right now it seems like the issue is specific to Windows Server 2019 Datacenter. I wonder if it's all Windows Server releases, or specific ones. I could try some other releaes, like 2016 for example:
```
 > gcloud compute images list --filter=windows | grep server
windows-server-2004-dc-core-v20210608                 windows-cloud      windows-2004-core                             READY
windows-server-2012-r2-dc-core-v20210608              windows-cloud      windows-2012-r2-core                          READY
windows-server-2012-r2-dc-v20210608                   windows-cloud      windows-2012-r2                               READY
windows-server-2016-dc-core-v20210608                 windows-cloud      windows-2016-core                             READY
windows-server-2016-dc-v20210608                      windows-cloud      windows-2016                                  READY
windows-server-2019-dc-core-for-containers-v20210608  windows-cloud      windows-2019-core-for-containers              READY
windows-server-2019-dc-core-v20210608                 windows-cloud      windows-2019-core                             READY
windows-server-2019-dc-for-containers-v20210608       windows-cloud      windows-2019-for-containers                   READY
windows-server-2019-dc-v20210608                      windows-cloud      windows-2019                                  READY
windows-server-20h2-dc-core-v20210608                 windows-cloud      windows-20h2-core                             READY
```

**Comment by jakubgs** [comment]
I tried building the node using adjusted [stack size](https://github.com/status-im/nimbus-eth2/blob/44f652f704f14c67e648f2b14dee652d44b858c5/config.nims#L33), both `16777216` and `33554432`, but neither worked.

**Comment by jakubgs** [comment]
Although, I think it might be getting further along and connecting to 10+ peers before the segmentation fault occurs.

**Comment by jakubgs** [comment]
Just to be sure I ran the same command via PowerShell rather than Bash that comes with Git and segfault still happens, although I don't actually see the segmentation fault message, and it just crashes after a while:

![image](https://user-images.githubusercontent.com/2212681/124114132-7c336c00-da6c-11eb-9884-123fb2813a35.png)

The exit code is `-1073741819`.

**Comment by jakubgs** [comment]
~~When I try to run beacon node via 64 bit WinDBG with the same flags it fails due to an assertion:~~
```
...\repo\beacon_chain\nimbus_signing_process.nim nimbus_signing_process
...\repo\vendor\nim-testutils\testutils\moduletests.nim main
...\repo\vendor\nimbus-build-system\vendor\Nim\lib\system\assertions.nim failedAssertImpl
...\repo\vendor\nimbus-build-system\vendor\Nim\lib\system\assertions.nim raiseAssert
...\repo\vendor\nimbus-build-system\vendor\Nim\lib\system\fatal.nim sysFatal
Error: unhandled exception: ...\repo\beacon_chain\nimbus_signing_process.nim(23, 12) `paramCount() == 2`  [AssertionError]
```
Scratch that, wrong binary.

**Comment by jakubgs** [comment]
The hell is this supposed to mean:
![image](https://user-images.githubusercontent.com/2212681/124132967-a6435900-da81-11eb-937c-a34584a07db5.png)
EDIT: Looks like the debugger is adding a newline because of how I pasted the arguments. Ugh.

**Comment by jakubgs** [comment]
Now it's forcing me to go step-by-step even though the crash hasn't surfaced yet:

```
(112c.13f0): Break instruction exception - code 80000003 (first chance)
ntdll!LdrpDoDebuggerBreak+0x30:
00007fff`0d24260c cc              int     3
0:000> g
ModLoad: 00007fff`08190000 00007fff`081c1000   C:\Windows\SYSTEM32\ntmarta.dll
ModLoad: 00007fff`08930000 00007fff`08997000   C:\Windows\system32\mswsock.dll
(112c.5c8): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
ntdll!RtlpWaitOnCriticalSection+0x87:
00007fff`0d1ce4eb ff4124          inc     dword ptr [rcx+24h] ds:00000000`00000024=????????
0:005> gn
(112c.5c8): Access violation - code c0000005 (!!! second chance !!!)
ntdll!RtlpWaitOnCriticalSection+0x87:
00007fff`0d1ce4eb ff4124          inc     dword ptr [rcx+24h] ds:00000000`00000024=????????
0:005> gh
(112c.5c8): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
ntdll!RtlpWaitOnCriticalSection+0x87:
00007fff`0d1ce4eb ff4124          inc     dword ptr [rcx+24h] ds:00000000`00000024=????????
0:005> g
(112c.5c8): Access violation - code c0000005 (!!! second chance !!!)
ntdll!RtlpWaitOnCriticalSection+0x87:
00007fff`0d1ce4eb ff4124          inc     dword ptr [rcx+24h] ds:00000000`00000024=????????
```

**Comment by jakubgs** [comment]
Or maybe it has crashed, I'm just not seeing the `segmentation fault` error.

**Comment by jakubgs** [comment]
The process doesn't go beyond `Starting metrics server`:

![image](https://user-images.githubusercontent.com/2212681/124134141-d3dcd200-da82-11eb-8711-f326dd6b72d3.png)

**Comment by jakubgs** [comment]
Interestingly enough, I dropped the metrics related flags and it kept going. Werid.

**Comment by jakubgs** [comment]
I've been watching it for 5 minutes now and it seems like it's not crashing, it already reached 25 peers:

![image](https://user-images.githubusercontent.com/2212681/124134810-7dbc5e80-da83-11eb-8496-64de1dee9df7.png)

It's possible that the crash is directly related to the metrics code.

**Comment by jakubgs** [comment]
I tested this without a debugger, and with `--metrics --metrics-address=0.0.0.0 --metrics-port=9200` flags it crashes:

![image](https://user-images.githubusercontent.com/2212681/124135510-28348180-da84-11eb-8bc6-3538edc9c796.png)

But without them it runs just fine:

![image](https://user-images.githubusercontent.com/2212681/124135421-17840b80-da84-11eb-89cd-f0a5a8b3040a.png)

So it does seem like the metrics code is the issue.

**Comment by jakubgs** [comment]
It appears that the call to the `/metrics` endpoint triggers the crash. Which explains why it didn't happen on Windows 10.

After triggering the crash by hand(since I disabled Consul) I go this from `kP` in debugger:
```
ModLoad: 00007fff`0b310000 00007fff`0b33e000   C:\Windows\System32\IMM32.DLL
ModLoad: 00007fff`0d030000 00007fff`0d09d000   C:\Windows\System32\WS2_32.dll
ModLoad: 00000000`64940000 00000000`64955000   C:\ProgramData\scoop\apps\gcc\current\bin\libwinpthread-1.dll
(1530.17a0): Break instruction exception - code 80000003 (first chance)
ntdll!LdrpDoDebuggerBreak+0x30:
00007fff`0d24260c cc              int     3
0:000> g
ModLoad: 00007fff`08190000 00007fff`081c1000   C:\Windows\SYSTEM32\ntmarta.dll
ModLoad: 00007fff`08930000 00007fff`08997000   C:\Windows\system32\mswsock.dll
(1530.17a0): Invalid handle - code c0000008 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
ntdll!KiRaiseUserExceptionDispatcher+0x3a:
00007fff`0d21381a 8b8424c0000000  mov     eax,dword ptr [rsp+0C0h] ss:00000000`14f97e70=c0000008
0:000> g
(1530.ad0): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
ntdll!RtlpWaitOnCriticalSection+0x87:
00007fff`0d1ce4eb ff4124          inc     dword ptr [rcx+24h] ds:00000000`00000024=????????
0:004> kP
 # Child-SP          RetAddr               Call Site
00 00000000`1ea6c9b0 00007fff`0d18449c     ntdll!RtlpWaitOnCriticalSection+0x87
01 00000000`1ea6ca20 00007fff`0d1843b0     ntdll!RtlpEnterCriticalSectionContended+0xdc
*** WARNING: Unable to verify timestamp for C:\Users\nimbus\beacon-node-prater-stable\bin\nimbus_beacon_node.exe
02 00000000`1ea6ca50 00000000`007d157a     ntdll!RtlEnterCriticalSection+0x40
03 00000000`1ea6ca80 00000000`007d2c52     nimbus_beacon_node+0x3d157a
04 00000000`1ea6cde0 00000000`00810b75     nimbus_beacon_node+0x3d2c52
05 00000000`1ea6cf90 00000000`0080f950     nimbus_beacon_node+0x410b75
06 00000000`1ea6e030 00000000`0081075a     nimbus_beacon_node+0x40f950
07 00000000`1ea6e500 00000000`00804d00     nimbus_beacon_node+0x41075a
08 00000000`1ea6e540 00000000`007f6809     nimbus_beacon_node+0x404d00
09 00000000`1ea6f330 00000000`006f8867     nimbus_beacon_node+0x3f6809
0a 00000000`1ea6f680 00000000`006fbde4     nimbus_beacon_node+0x2f8867
0b 00000000`1ea6f800 00000000`0080f6d4     nimbus_beacon_node+0x2fbde4
0c 00000000`1ea6f8c0 00000000`005acee8     nimbus_beacon_node+0x40f6d4
0d 00000000`1ea6fc00 00000000`0059dc90     nimbus_beacon_node+0x1acee8
0e 00000000`1ea6fed0 00007fff`0ae07974     nimbus_beacon_node+0x19dc90
0f 00000000`1ea6ff30 00007fff`0d1ca2f1     KERNEL32!BaseThreadInitThunk+0x14
10 00000000`1ea6ff60 00000000`00000000     ntdll!RtlUserThreadStart+0x21
```

**Comment by arnetheduck** [comment]
fixed in 1.4.1
